### PR TITLE
Add ingest info to Cluster Stats

### LINKED ISSUE #46146: Add a summary of IngestStats to ClusterStatsNodes
It would be nice to see a unified view of which processors are being used by the cluster in the cluster-stats response. Right now, ingest-stats is present in node stats, but it is left out of cluster-stats. This issue is open to discuss and track if and how it makes sense to expose these details in the `ClusterStatsNodes` part of the `ClusterStatsResponse`

Some information that would be interesting to provide that is missing

- number of pipelines present
- set of processor types used

### PR DESCRIPTION
**Pull request by talevy**
This PR enhances the ClusterStatsNodes response to include global 
processor usage stats on a per-processor basis.

example output:

```
...
    "processor_stats": {
      "gsub": {
        "count": 0,
        "failed": 0
        "current": 0
        "time_in_millis": 0
      },
      "script": {
        "count": 0,
        "failed": 0
        "current": 0,
        "time_in_millis": 0
      }
    }
...
```

The purpose for this enhancement is to make it easier to collect stats on how specific processors are being used across the cluster beyond the current per-node usage statistics that currently exist in node stats.

Closes #46146.

### CODE REVIEW

**Comment by talevy** [comment]
run elasticsearch-ci/bwc

File: server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java [hunk 1]
```diff
@@ -690,4 +696,56 @@ public XContentBuilder toXContent(final XContentBuilder builder, final Params pa
...
+            SortedMap<String, long[]> stats = new TreeMap<>();
+            for (NodeStats nodeStat : nodeStats) {
+                if (nodeStat.getIngestStats() != null) {
+                    for (Map.Entry<String,
+                            List<org.elasticsearch.ingest.IngestStats.ProcessorStat>> processorStats : nodeStat.getIngestStats()
+                            .getProcessorStats().entrySet()) {
+                        pipelineIds.add(processorStats.getKey());
+                        for (org.elasticsearch.ingest.IngestStats.ProcessorStat stat : processorStats.getValue()) {
+                            stats.compute(stat.getType(), (k, v) -> {
+                                if (v == null) {
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: should we also add `ingestTimeInMillis` here?

File: server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java [hunk 2]
```diff
@@ -690,4 +696,56 @@ public XContentBuilder toXContent(final XContentBuilder builder, final Params pa
...
+                                    v[1] += stat.getStats().getIngestFailedCount();
+                                    return v;
+                                }
+                            });
+                        }
+                    }
+                }
+            }
+            this.pipelineCount = pipelineIds.size();
+            this.stats = Collections.unmodifiableMap(stats);
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: use Map#copyOf(...) here instead?

**Review by martijnvg — COMMENTED** [review]
This looks good! I left a few small comments.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java (hunk 1 above)
Comment (talevy) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: hmm. I suppose you are right. I was only interested in adding fields that I already know usages of, but in order to be more complete and have feature-parity with the node stats API, both `ingestTimeInMillis` and `ingestCurrent` should be added. is that what you mean?

File: server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java [hunk 3]
```diff
@@ -64,6 +65,7 @@
     private final NetworkTypes networkTypes;
     private final DiscoveryTypes discoveryTypes;
     private final PackagingTypes packagingTypes;
+    private IngestStats ingestStats;
```
[the remark this answers is not in the document]
Comment (talevy) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: yup!

File: server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java (hunk 2 above)
Comment (talevy) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: yup!

File: server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java (hunk 1 above)
Comment (martijnvg) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: I think `ingestTimeInMillis` is an interesting metric to capture.
I'm not so sure about `ingestCurrent`. In any way I think for feature parity it is good to capture both.

Comment (talevy) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: agreed. thanks for bringing it up. I'll update to include these

Comment (talevy) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: updated. also found a mismatch in the naming of the `failed` field. it was `failed_count` in this PR for cluster stats, while `failed` in node stats.

**Review by martijnvg — APPROVED** [review]
LGTM!

**Pull request merged**